### PR TITLE
Settings frontend new css

### PR DESCRIPTION
**Pull request by twigleingrid**

### CODE REVIEW

**Comment by The-Compiler** [comment]
Yay! I think this is a big improvement over the old settings page:

image

image

cc @iggy who added the original one, many years ago (in fact, the second pull request in this repository! #2)

File: qutebrowser/html/settings.html [hunk 1]
```diff
@@ -37,18 +141,38 @@
...
+	        name="{{ option.name }}" value="{{ value }}"
+	        onclick="cset('{{ option.name }}', this.value)">
+	      <label for="input-{{ option.name }}-{{ loop.index0 }}">
+	        {{ value }}
+	      </label>
+	    </div>
+	    {% if confget(option.name) == value %}
+	      <script type="text/javascript">
+	        checkInput('{{ option.name}}', {{ loop.index0 }})
+	      </script>
```
Comment (The-Compiler) [inline qutebrowser/html/settings.html:163]: I'm wondering why this JS snippet is needed - couldn't you just add the [`checked` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#attr-checked) directly to the `<input>` element above by doing the `{% if ... %}` inside the `<input>` attributes above?

**Review by The-Compiler — CHANGES_REQUESTED** [review]
Not a full in-depth review yet - just something that caught my eye after a quick look.

**Comment by iggy** [comment]
Looks good (aesthetically, I've basically forgotten every bit of the code I originally wrote, so can't really comment on that). I'm pretty sure I do at least remember that at the time webkit didn't support a lot of the fancier CSS stuff. Glad someone finally took a critical eye to it.

**Comment by mschilli87** [comment]
For me it is not clear which of the binary options (true/false, webkit/webengine) is selected based on the coloring only.
This is no objection from my side against the changes suggested here since the left hand side spells out the current value but I figured I'd leave the feedback here anyways. :wink:

**Comment by The-Compiler** [comment]
Actually, the value on the left-hand side is static, i.e. it doesn't update when changing the value from the web UI. I'm not entirely sure where it's there - I'd be in favor of removing it.

As for making it clearer what the selected value is, maybe a lighter grey for unselected values would already make things clearer? I don't know - maybe @twigleingrid has another idea.

File: qutebrowser/html/settings.html (hunk 1 above)
Comment (twigleingrid) [reply qutebrowser/html/settings.html:163]: Yep, that's true, I am still getting used to Jinja templates :blush:

**Comment by twigleingrid** [comment]
> As for making it clearer what the selected value is

Yep, I was also thinking about making the color less strong, how about something like this? 
qute-grey


> Actually, the value on the left-hand side is static, i.e. it doesn't update when changing the value from the web UI. I'm not entirely sure where it's there - I'd be in favor of removing it. 

It is a bit confusing that it doesn't update when changing the value. The advantage I see to that is: one can see the previous setting if one changed something by mistake. But maybe that should be displayed in a different, less confusing way. Maybe a revert to the default_value button?

**Comment by The-Compiler** [comment]
I guess having it in a more faint color would help. Perhaps even something grey, as grey stands (I hope) somewhat universally for "unselected"?

image

(that's `#dddddd` with text color `#666666`)

As for clearing the value, I suppose a "reset to default" button which actually unsets the value would be great. There's a difference between setting a value to its default and not having it set at all - when unsetting it, you'd e.g. get new defaults when qutebrowser changes them in an update. But perhaps this should be separate from those initial changes?

**Comment by twigleingrid** [comment]
> Perhaps even something grey, as grey stands (I hope) somewhat universally for "unselected"?
Haha, ok, let's hope so! :smile: 

> But perhaps this should be separate from those initial changes?
Yep, the button was an idea for the future, let's speak about it in more detail elsewhere. 

> Not a full in-depth review yet
Looking forward for you other ideas!

File: qutebrowser/html/settings.html [hunk 2]
```diff
@@ -13,22 +13,112 @@
...
+    border: 2px solid #7a589ea6;
+}
+
+/* Hides the radio button */
+input[type="radio"] {
+    position: absolute;
+    opacity: 1%;
+    width: min-content;
+    margin: 0;
+}
```
Comment (The-Compiler) [inline qutebrowser/html/settings.html:68]: I'm a bit confused on how this works - wouldn't something like `display: none` suffice?

FWIW this also doesn't seem to work properly with an older QtWebEngine version - you should be able to test with `tox -e py39-pyqt512 --notest` (adjust `py39` if you're using a different Python version), followed by `.tox/py39-pyqt512/bin/python3 -m qutebrowser --temp-basedir :set`:

image

(looks like `opacity:` in CSS [was added with Chrome 78](https://caniuse.com/mdn-css_properties_opacity_percentages), while Qt 5.12 is based on 69)

**Review by The-Compiler — CHANGES_REQUESTED** [review]
One remaining issue with how you're hiding the radio buttons, then I think this is good to go!

File: qutebrowser/html/settings.html (hunk 2 above)
Comment (twigleingrid) [reply qutebrowser/html/settings.html]: Thank you for the review!

`display: none` breaks the `f` hinting :(
breaks_f_hint

But `-webkit-appearance: none;` seems to work with the `f` hinting, yay! 
Also for the older version, as it is partially supported for [Chrome 4-81](https://caniuse.com/?search=-webkit-appearance)

I tried to write a few more beautiful comments, I hope that will help!

**Comment by The-Compiler** [comment]
Looks great now, merged - thank you! :+1:

**Pull request merged**